### PR TITLE
Get TenureByID to return ETag headers

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/jira/software/projects/MTTL/boards/47?selectedIssue=MTTL-1509)

## Describe this PR

Updated the GetByID endpoint to return ETag headers
#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

Are there any next steps that need to addressed after merging this PR? Add them here.

### CODE REVIEW

File: TenureInformationApi/V1/Controllers/TenureInformationController.cs [hunk 1]
```diff
@@ -50,7 +52,9 @@ public async Task<IActionResult> GetByID([FromRoute] TenureQueryRequest query)
         {
             var result = await _getByIdUseCase.Execute(query).ConfigureAwait(false);
             if (result == null) return NotFound(query.Id);
-            return Ok(result);
+
+            HttpContext.Response.Headers.Add(HeaderConstants.ETag, result.VersionNumber.ToString());
```
Comment (amidomichael) [inline TenureInformationApi/V1/Controllers/TenureInformationController.cs:56]: could this be surrounded by quotes, e.g. `"0"`

File: TenureInformationApi/V1/Controllers/TenureInformationController.cs [hunk 2]
```diff
@@ -50,7 +52,9 @@ public async Task<IActionResult> GetByID([FromRoute] TenureQueryRequest query)
         {
             var result = await _getByIdUseCase.Execute(query).ConfigureAwait(false);
             if (result == null) return NotFound(query.Id);
-            return Ok(result);
+
+            HttpContext.Response.Headers.Add(HeaderConstants.ETag, result.VersionNumber.Value.ToString());
```
Comment (mikemoody-amido) [inline TenureInformationApi/V1/Controllers/TenureInformationController.cs:56]: Shouldn't this be...
```
HttpContext.Response.Headers.Add(HeaderConstants.ETag, $"\"{result.VersionNumber.Value}\"");
```

Comment (amidomichael) [reply TenureInformationApi/V1/Controllers/TenureInformationController.cs:56]: or `new EntityTagHeaderValue($"\"{result.VersionNumber.Value}\"", false).ToString()`

Comment (humulla) [reply TenureInformationApi/V1/Controllers/TenureInformationController.cs:56]: I've just tried Mike's suggestion and it looks like it has worked!

**Review by amidomichael — APPROVED** [review]

**Pull request merged**